TG-2387 Implemented tax and basic sizing

paulkennethkent opened the pull request:
Tax is added to all products except when the object size and category match the exempt criteria. Sizing implementation can be improved but I think this should be considered when we work on the stock control feature which will require big changes to the models.

JohnLBergqvist commented on src/main/resources/templates/ProductPage.html:
```diff
@@ -13,6 +13,11 @@
+                      <select name="size" required="required">
+                        <option value="">-- Please select a size --</option>  
+                        <option value="Child">Child</option>
+                        <option value="Adults">Adult</option>
```
The value here should be `Adult`, no?

jgwilson42 reviewed (changes requested):
Please could you replace BigDecimal with double?

peterschrammel commented on src/main/java/com/diffblue/demo/ecommerce/models/Order.java:
```diff
@@ -57,7 +57,7 @@
    * @param shippingAddress - shipping address
    */
-  public Order(String orderRef, String firstName, String lastName,
+  public Order(String orderRef, String firstName, String lastName, 
```
trailing whitespace

peterschrammel commented on src/main/java/com/diffblue/demo/ecommerce/models/OrderItem.java:
```diff
@@ -1,6 +1,6 @@
 package com.diffblue.demo.ecommerce.models;
 
-// Copyright 2016-2017 DiffBlue limited. All rights reserved.
+// Copyright 2016-2018 DiffBlue limited. All rights reserved.
```
Diffblue (all files)

peterschrammel commented on src/main/java/com/diffblue/demo/ecommerce/models/Cart.java:
```diff
@@ -6,11 +6,16 @@
 import java.util.HashMap;
 import java.util.Map;
 
+
```
spurious newlines

peterschrammel commented on src/main/java/com/diffblue/demo/ecommerce/models/Cart.java:
```diff
@@ -6,11 +6,16 @@
   private Map<Product, Integer> cartItems;
   private BigDecimal subtotal;
   private BigDecimal shipping;
+  private BigDecimal tax;
```
Shouldn't this be double for now?

peterschrammel commented on src/main/java/com/diffblue/demo/ecommerce/models/Cart.java:
```diff
@@ -6,8 +6,6 @@
 import java.util.Map;
 
-
-
```
In a proper dev workflow these fixes should be squashed into the commit that created the problem in the first place (if this commit is in the same PR).

peterschrammel commented on src/main/resources/templates/ProductPage.html:
```diff
@@ -16,7 +16,7 @@
                         <option value="">-- Please select a size --</option>  
                         <option value="Child">Child</option>
-                        <option value="Adults">Adult</option>
+                        <option value="Adult">Adult</option>
```
squash this into the original commit

peterschrammel reviewed (changes requested):
Commit 
"Merge branch 'master' into cart-tax" should be removed by rebasing the branch on master.

peterschrammel commented:
Note: In a proper dev workflow these fixes should be squashed into the commit that created the problem in the first place (if this commit is in the same PR). I'm reviewing PRs commit by commit - if something is fixed up in a later commit I will still have put a comment on the earlier commit which then appears as 'outdated'.

paulkennethkent committed:
Implemented tax and basic sizing
A basic tax rate of 20 percent is added to all items expect to child apparel. The sizing implementation is basic and will need to be improved in a future PR.

peterschrammel reviewed (approved):
LGTM
